utils/text_formatting.tsx

```diff
@@ -262,6 +271,16 @@ export function formatText(
         output = `<span class="all-emoji">${output.trim()}</span>`;
     }
 
+    if (options.postId) {
+        // unwrap the output from the closing p tag and add a span that will serve as a
+        // palceholder for `messageToHtmlComponent` function later on
+        if (output.endsWith('</p>')) {
+            output = `${output.slice(0, -4)}<span data-edited-post-id='${options.postId}'></span></p>`;
```

Comment:
0/5 to avoid magic numbers, I would prefer something like  `output.slice(0, -'</p>'.length)`

Reply:
seems a bit strange to do it like this. It looks overcomplicated. I did write a comment that should explain what is happening. IMO I would leave it like this for now.